### ISSUE TITLE
Unable to load schema error when running VSCode with SSH remote extension

### ISSUE THREAD

**Issue by ntompson** [keesschollaart81/vscode-home-assistant#2602]
Extension runs in:
- [ ] VS Code for Desktop 
- [X] Hassio Add-on (either SSH add-on or VSCode Remote add-on)
- [ ] Visual Studio Online
- [ ] Other: 

I'm running VS Code on:
- [ ] Windows 
- [X] Mac 
- [ ] Linux 

I'm accessing my files:
- [ ] From local disk 
- [X] Via remote SSH 
- [ ] Smb
- [ ] Other:  

I've successfully connected VSCode to my HA config using the SSH Remote extension. I have successfully connected to either the core SSH add-on or the edge VSCode Remote add-on. Everything seems to work well, including YAML linting, auto-completion of entity names, icon matching, all the rest of it. However, for either target add-on, every YAML file is marked with the same error. Here's an example:

<img width="970" alt="Screenshot 2023-04-05 at 12 18 06 pm" src="https://user-images.githubusercontent.com/32885870/229964046-9b9ada94-a326-4876-be91-60f80cc7019b.png">

The extension is running at the remote end (ie within either add-on), not on my local VSCode. The extension version deployed today is `v1.35.1`.

I don't know whether it is relevant, but I get a "can't find server error" if I try to navigate to http://schemas.home-assistant.io in a browser. `curl` also complains that `curl: (6) Could not resolve host: schemas.home-assistant.io`.

**Comment by krwill16** [comment]
I am seeing the same thing, but I presumed I was missing something on my end. `¯\_(ツ)_/¯`

**Comment by cyndane31** [comment]
Seeing the same behavior in VS Code desktop on linux, accessing files via mounted Samba share.  I am NOT getting this error in the Hassio VSCode Remote add-on.

**Comment by krwill16** [comment]
I was accessing via VS Code for MacOS, but also accessing the config files via a Samba share. That would be interesting if that was relevant. I don't know much about how the schema type logic works, but it seems like that shouldn't matter? I'm not one to say for sure. 🙂

**Comment by arychj** [comment]
I think the `schemas.home-assistant.io` in the URL is just a namespace used for the generated schemas, and it looks like the extension is having issues locating the schema in its cache. Relatedly, it looks like dependabot bumped the package used to generate the schemas just before the 1.35.1 release was cut: https://github.com/keesschollaart81/vscode-home-assistant/pull/2598.

I rolled back the extension to 1.35.0 and it seems to have resolved the problem.

**Comment by krwill16** [comment]
Was tinkering with the extention myself. Few interesting notes; 
1) As mentioned to check in the wiki, when set to just YAML all checks out - so not a YAML parsing issue. (did not think it was)
2) Still looking it over but there is some logic in that 56 version of `typescript-json-schema` about skipping files that was added. I am trying to see if perhaps this changed how the generation is occurring in this plugin. https://github.com/YousefED/typescript-json-schema/compare/v0.55.0...v0.56.0 
3) Along with that the `typescript-json-schema` dependency, seems to pull in a new `typescript` version, ~4.9.5, which talks about a breaking fix to imports as well as some promise resolving changes. https://github.com/microsoft/TypeScript/wiki/Breaking-Changes#typescript-49. That is interesting because if you open the extension locally it seems to have errors (for me at least) that seem related. (preventing starting extension locally)

Little outside my depth - but find this interesting so may keep poking around. 😄

**Comment by daballiemo** [comment]
Exact same setup and behaviour here. Rolling back to 1.35.0 seems to solve the issue.

**Comment by regularguy01** [comment]
+1 same issue

**Comment by sjf-control** [comment]
I have the same problem.  Opening any yaml file in "Studio Code Server" generates the error "Unable to load schema from 'http://schemas.home-assistant.io/configuration': No content." The last part varies depending on the file's name
I'm not clear, though, on what you are rolling back to 1.35.0.  The studio code server is version 5.5.7

**Comment by user34756361233** [comment]
This problem is with VS-Code-Assistant which is currently at version 1.35.1; not VS Code....

![Screenshot 2023-04-21 at 19 12 29](https://user-images.githubusercontent.com/16369502/233695867-8ae16d5c-8eea-4c6e-9baa-7f9ad4b85c3d.png)

**Comment by sjf-control** [comment]
> This problem is with VS-Code-Assistant which is currently at version 1.35.1; not VS Code....
> 
Ah! And I do have that installed, too.  Thanks for the clarification.

**Comment by KreNtal** [comment]
How can I rollback to previous version? Because Install another version.. is greyed out

**Comment by marleyjaffe** [comment]
You can restore from a previous backup to restore to a prior version. I don't believe there is any other way.

**Comment by user34756361233** [comment]
You are doing this from within VS Code itself? And you the same as I posted before https://user-images.githubusercontent.com/16369502/233695867-8ae16d5c-8eea-4c6e-9baa-7f9ad4b85c3d.png

**Comment by excalbian** [comment]
> > How can I rollback to previous version? Because Install another version.. is greyed out
> 
> I have the same question.

If you're using the VS Code add-on within Home Assistant the plugin is bundled and you can't change it. The only option seems to be [restoring the add-on from a backup](https://github.com/hassio-addons/addon-vscode/issues/611#issuecomment-1518710567) until this is fixed in the plugin.

**Comment by Bascht74** [comment]
Hi, I do have the same problem. Is there a fix on it's way?
The issue is now open more than a month?
I am using the HA add on, where you cannot change the version of the extension...

**Comment by oxtn** [comment]
For those that don't have or can't restore a partial backup of the VSCode add-on, you can run the following

1. Stop the VSCode addon
2. Open the Terminal through the Advanced SSH add-on w/ protection mode disabled
3. Run this script to pull the old Docker image
```
arch="$(ha su info --raw-json | jq -r '.data.arch')"
docker pull "ghcr.io/hassio-addons/vscode/${arch}:5.5.6"
docker tag "ghcr.io/hassio-addons/vscode/${arch}:5.5.6" "ghcr.io/hassio-addons/vscode/${arch}:5.5.7"
```
4. Restart the VSCode addon

**Comment by JONES6137** [comment]
> For those that don't have or can't restore a partial backup of the VSCode add-on, you can run the following
> 
> 1. Stop the VSCode addon
> 2. Open the Terminal through the Advanced SSH add-on w/ protection mode disabled
> 3. Run this script to pull the old Docker image
> 
> ```
> arch="$(ha su info --raw-json | jq -r '.data.arch')"
> docker pull "ghcr.io/hassio-addons/vscode/${arch}:5.5.6"
> docker tag "ghcr.io/hassio-addons/vscode/${arch}:5.5.6" "ghcr.io/hassio-addons/vscode/${arch}:5.5.7"
> ```
> 
> 4. Restart the VSCode addon

Yay this worked!  Thanks so much for sharing. Perfect timing as I was restructuring my configs.  :)

**Comment by frenck** [comment]
I would strongly advise against that. You mess up the tracking and version management of the Supervisor and will lead to incorrect behavior.

Restore a backup. Each time you update an add a backup is made. Easy clean and fast. 

One should not mess with docker there, nor should you ever instruct people to do so. They might get themselves into positions they won't be able to get themselves out of...

../Frenck

**Comment by Virenbar** [comment]
Some users may not have backups because they didn't use the add-on before version 5.5.7, such as myself.
But I fixed the bug in a different way, using the installation of a "new" version of the extension.
[I described the solution on the HA forum.](https://community.home-assistant.io/t/unable-to-load-schema-from-http-schemas-home-assistant-io-integration-automation-no-content-yaml-768/557688/35?u=virenbar) 
In short, I modified the previous version of the extension from 1.35.0 to 1.35.2 and installed it.

**Comment by kpurintun** [comment]
> I would strongly advise against that. You mess up the tracking and version management of the Supervisor and will lead to incorrect behavior.
> 
> Restore a backup. Each time you update an add a backup is made. Easy clean and fast.
> 
> One should not mess with docker there, nor should you ever instruct people to do so. They might get themselves into positions they won't be able to get themselves out of...
> 
> ../Frenck

This has been in a not good state since Apr 3rd..  is there an official fix coming soon?  or official roll back?  Seems like I would have to roll back (partial) to April 21st to get back to 5.5.6.  rolling back that far is a bit scary.   If it is a 'partial backup' will it just reload the addon part of the backup?

Went to troubleshoot some issue with a rain gauge and opened up a real ugly can of ants..  not even sure what it means to not be able to load the schemas..  some people said that the files opened 'would not be tracked once opened'.  I just opened my sensors and templates..  so that pretty much sucks.  not totally sure what 'tracked means'

Kyle

**Comment by ntompson** [comment]
This seems to be impacting the "Studio Code Server" add-on now (as of upgrading HA from 2023.4.x to 2023.5.3), where previously it was only impacting the use of a local VSCode application connecting via the SSH remote extension. I had stopped using the SSH remote extension and went back to the Studio Code Server add-on, but now there doesn't seem to be any way to avoid the error - it must be impacting everyone. 

None of us seem to have much idea about what the problem is or how to fix it...??

**Comment by JONES6137** [comment]
> I would strongly advise against that. You mess up the tracking and version management of the Supervisor and will lead to incorrect behavior.
> 
> Restore a backup. Each time you update an add a backup is made. Easy clean and fast. 
> 
> One should not mess with docker there, nor should you ever instruct people to do so. They might get themselves into positions they won't be able to get themselves out of...
> 
> ../Frenck

Oh yikes. Thank you for the heads up. Is there a way I can remedy this other than restoring from a backup?

**Comment by PasekJ** [comment]
I just uninstalled VS code and stopped using it

**Comment by DarthDestroyer** [comment]
Its closed, but i found a workaround someone posted in the community...

https://community.home-assistant.io/t/unable-to-load-schema-from-http-schemas-home-assistant-io-integration-automation-no-content-yaml-768/557688/35

Just edit and upload the Package - everything works as intended - no need to rollback a backup

**Comment by TonyM1958** [comment]
Looks like this is fixed. I had the error and it went away when I updated to Studio Code Server 5.6.0 for anyone following this thread.

**Comment by dannytsang** [comment]
It's fixed for me 👍
